rosbag2_compression/include/rosbag2_compression/sequential_compression_writer.hpp

```diff
@@ -104,9 +144,27 @@ class ROSBAG2_COMPRESSION_PUBLIC SequentialCompressionWriter
    */
   virtual void setup_compression();
 
+  /**
+   * Initializes a number of threads to do file or message compression equal to the
+   * value of the compression_threads parameter.
+   */
+  virtual void setup_compressor_threads();
+
+  /**
+   * Signals all compressor threads to stop working and then waits for them to exit.
+   */
+  virtual void stop_compressor_threads();
+
 private:
   std::unique_ptr<rosbag2_compression::BaseCompressorInterface> compressor_{};
   std::unique_ptr<rosbag2_compression::CompressionFactory> compression_factory_{};
+  std::queue<std::shared_ptr<rosbag2_storage::SerializedBagMessage>> compressor_message_queue_;
+  std::queue<std::string> compressor_file_queue_;
+  std::mutex compressor_mutex_;
+  std::vector<std::thread> compression_threads_;
+  bool compression_is_running_{false};
```

Comment:
should this maybe be `atomic_bool` - is there any danger of race condition in its usage? not sure exactly, probably not because it's only read by the worker threads, not written

Reply:
I don't think it's a problem, but it can't hurt to change it, so I'll do so.